File: java/Samples/BiometricStandards/CS/IIRecordSampleCS/IIRecordOptionsForm.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Neurotec.Biometrics.Common;
using Neurotec.Biometrics.Standards;

namespace Neurotec.Samples
{
	public partial class IIRecordOptionsForm : BdifOptionsForm
	{
		public IIRecordOptionsForm()
		{
			InitializeComponent();
			var versions = new List<StandardVersion>();
			versions.Add(new StandardVersion(BdifStandard.Ansi, IIRecord.VersionAnsi10, "ANSI/INCITS 379-2004"));
			versions.Add(new StandardVersion(BdifStandard.Iso, IIRecord.VersionIso10, "ISO/IEC 19794-6:2005"));
			versions.Add(new StandardVersion(BdifStandard.Iso, IIRecord.VersionIso20, "ISO/IEC 19794-6:2011"));
			StandardVersions = versions.ToArray();
		}

		public override uint Flags
		{
			get
			{
				uint flags = base.Flags;
				if (cbProcessFirstIrisImageOnly.Checked)
					flags |= IIRecord.FlagProcessIrisFirstIrisImageOnly;

				return flags;
			}
			set
			{
				if ((value & IIRecord.FlagProcessIrisFirstIrisImageOnly) == IIRecord.FlagProcessIrisFirstIrisImageOnly)
					cbProcessFirstIrisImageOnly.Checked = true;
				base.Flags = value;
			}
		}

		protected override void OnModeChanged()
		{
			base.OnModeChanged();

			switch (Mode)
			{
				case BdifOptionsFormMode.New:
					cbProcessFirstIrisImageOnly.Enabled = false;
					break;
			}
		}
	}
}

File: java/Samples/BiometricStandards/VB/IIRecordSampleVB/IIRecordOptionsForm.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports Neurotec.Biometrics.Standards

Partial Public Class IIRecordOptionsForm
	Inherits BdifOptionsForm
	Public Sub New()
		InitializeComponent()
		Dim versions = New List(Of StandardVersion)()

		Dim ansiV1 As StandardVersion = New StandardVersion()
		ansiV1.Standard = BdifStandard.Ansi
		ansiV1.Version = IIRecord.VersionAnsi10
		ansiV1.StandardName = "ANSI/INCITS 379-2004"

		Dim isoV1 As StandardVersion = New StandardVersion()
		isoV1.Standard = BdifStandard.Iso
		isoV1.Version = IIRecord.VersionIso10
		isoV1.StandardName = "ISO/IEC 19794-6:2005"

		Dim isoV2 As StandardVersion = New StandardVersion()
		isoV2.Standard = BdifStandard.Iso
		isoV2.Version = IIRecord.VersionIso20
		isoV2.StandardName = "ISO/IEC 19794-6:2011"

		versions.Add(ansiV1)
		versions.Add(isoV1)
		versions.Add(isoV2)

		StandardVersions = versions.ToArray()
	End Sub

	Public Overrides Property Flags() As UInteger
		Get
			Dim flags_Renamed As UInteger = MyBase.Flags
			If cbProcessFirstIrisImageOnly.Checked Then
				flags_Renamed = flags_Renamed Or IIRecord.FlagProcessIrisFirstIrisImageOnly
			End If

			Return flags_Renamed
		End Get
		Set(ByVal value As UInteger)
			If (value And IIRecord.FlagProcessIrisFirstIrisImageOnly) = IIRecord.FlagProcessIrisFirstIrisImageOnly Then
				cbProcessFirstIrisImageOnly.Checked = True
			End If
			MyBase.Flags = value
		End Set
	End Property

	Protected Overrides Sub OnModeChanged()
		MyBase.OnModeChanged()

		Select Case Mode
			Case BdifOptionsFormMode.New
				cbProcessFirstIrisImageOnly.Enabled = False
		End Select
	End Sub
End Class
